shared/actions/engine/helper.js

```diff
@@ -41,9 +41,9 @@ type RpcRunResult = Finished | FluxTypes.NoErrorTypedAction<'@@engineRPCCall:bai
 
 function _sagaWaitingDecorator(rpcNameKey, saga) {
   return function* _sagaWaitingDecoratorHelper(...args: any) {
-    yield put(Creators.waitingForRpc(rpcNameKey, false))
-    yield call(saga, ...args)
     yield put(Creators.waitingForRpc(rpcNameKey, true))
+    yield call(saga, ...args)
+    yield put(Creators.waitingForRpc(rpcNameKey, false))
```

Comment:
We want to set waiting to true before we call the RPC, and then to false after, right?

Reply:
i actually think the old code is correct. its part of the back and forth so we're actually replying . @MarcoPolo should double check